### ISSUE TITLE
U/P authentication not supported in .NET Standard in MSAL 3.0.0-preview

### ISSUE THREAD

**Issue by cormacpayne** [AzureAD/microsoft-authentication-library-for-dotnet#938]
**Which Version of MSAL are you using ?**
Note that to get help, you need to run the latest version. Preview version are also ok. 
For ADAL, please log issues to https://github.com/AzureAD/azure-activedirectory-library-for-dotnet

MSAL 3.0.0-preview

**Platform**

.NET Standard 2.0

**What authentication flow has the issue?**
* Desktop / Mobile
    * [ ] Interactive
    * [ ] Integrated Windows Auth
    * [x] Username Password
    * [ ] Device code flow (browserless)
* Web App
    * [ ] Authorization code
    * [ ] OBO
* Web API
    * [ ] OBO

Other? - please describe;

**Is this a new or existing app?**

Existing app in production, upgraded to latest version of MSAL.

**Repro**

```csharp
var scopes = new string [] { resource + "/user_impersonation" };
var publicClient = PublicClientApplicationBuilder.Create(clientId)
                                                                          .WithAuthority(authority)
                                                                          .Build();
// Token cache code here
var response = publicClient.AcquireTokenByUsernamePassword(scopes, userId, password)
                                          .ExecuteAsync();
```

**Expected behavior**
A clear and concise description of what you expected to happen (or code).

U/P authentication should successfully authenticate the user when running in a project targeting .NET Standard.

**Actual behavior**
A clear and concise description of what happens, e.g. exception is thrown, UI freezes  

The following `PlatformNotSupportedException` is thrown:

`Username Password is only supported on NetFramework and .NET Core.`

**Possible Solution**

**Additional context/ Logs / Screenshots**
Add any other context about the problem here, such as logs and screebshots. Logging is described at https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/logging

U/P authentication was previously working in MSAL 2.7.0.

**Comment by jmprieur** [comment]
See also https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/882

**Comment by cormacpayne** [comment]
Just confirmed this was fixed in `2.7.1-internal06401dev3x`, thanks! 😀

**Comment by jmprieur** [comment]
Thanks for the update @cormacpayne
